### PR TITLE
As a provider, I can view order details

### PR DESCRIPTION
**Pull request by laurawhalin**
This is the second small step(ha!) toward getting orders up and running. With this merge, it is now possible for a supplier to see their open orders and to view the details of that order. The orders and line items are not yet filtered out by supplier, which is the next step. The final step will be implementing the orders on the user side.

Please take a close look at the ability file. I am not sure if I changed the permissions by removing the reference to the user_id, but since the orders table no longer has a user_id column, this was breaking the app.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def initialize(user)
 #
     else
       can :read, User, id: user.id
-      can :read, Order, user_id: user.id
+      can :read, Order#, user_id: user.id  not sure if I should replace this with something else that identifies the user?
```
Comment (laurawhalin) [inline app/models/ability.rb:14]: I re-pushed without this comment in here, but it's still showing?
But more importantly, this is still a question I have. Is it ok to remove the user_id part of that and still have the permissions work the way they should?

File: app/views/suppliers/orders/_delivery_method.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      <%= @shipping_address.line_one %></br>
+      <%= @shipping_address.line_two %></br>
+      <%= @shipping_address.city %></br>
+      <%= @shipping_address.state %></br>
+      <%= @shipping_address.postal_code %></br>
+      <%= @shipping_address.country %>
+    </p>
+<% elsif @coordinates %>
+  <h5>Delivery Coordinates:</h5>
+<% else %>
```
Comment (laurawhalin) [inline app/views/suppliers/orders/_delivery_method.html.erb:13]: I will fill in code for the delivery coordinates once I figure that piece out, probably not until I get to the user side of orders. I'm planning on creating another table for coordinates that only belongs to orders.

File: app/views/suppliers/orders/_delivery_method.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  <h5>Delivery Method:</h5>
+    <p>
+      Pick-up
+    </p>
+<% end %>
```
Comment (laurawhalin) [inline app/views/suppliers/orders/_delivery_method.html.erb:18]: This entire partial view is very ugly on the web. It needs front-end lovin'! :heart_eyes:

File: app/views/suppliers/orders/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  <h5>Order Details</h5>
+  <tr>
+    <th>Item</th><th>Quantity</th><th>Price</th>
+  </tr>
+
+  <% @order.orders_items.each do |order_item|%>
+  <tr>
+    <td><%= link_to order_item.item.title, item_path(order_item.item) %></td>
+    <td><%= order_item.quantity %></td>
+    <td><%= number_to_currency(order_item.subtotal.to_f / 100.00) %></td>
```
Comment (laurawhalin) [inline app/views/suppliers/orders/show.html.erb:28]: First thing I'm going to do on my next branch is replace this with the helper method Alex created for formatting price.

File: app/views/suppliers/orders/_delivery_method.html.erb (hunk 3 above)
Comment (rshea303) [inline app/views/suppliers/orders/_delivery_method.html.erb:18]: I will take a look at the front end for this partial

File: app/views/suppliers/orders/_delivery_method.html.erb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<% if @shipping_address %>
+  <h5>Shipping Address:</h5>
+    <p>
+      <%= @shipping_address.line_one %></br>
```
Comment (rshea303) [inline app/views/suppliers/orders/_delivery_method.html.erb:4]: I've seen the 'break' tag written with the forward slash after the letter 'r' and I've seen it without a forward slash, but never in front.  Does it work the same way as a regular break when the slash is in the front?

**Comment by bmrsny** [comment]
I like the idea of a join table for the addresses.

**Pull request merged**